service/timestreamwrite/api_client.go

```diff
@@ -24,6 +29,8 @@ const ServiceAPIVersion = "2018-11-01"
 // Write.
 type Client struct {
 	options Options
+
+	endpointCache *discovery.EndpointCache
```

Comment:
We should consider making this `endpointCache` member an interface type not hard dependency on `discovery.EndpointCache` implementation.  A Default resolver could provide the value similar to HTTP client but this probably should be an interface for similar reasons as HTTP client is interface.